App/Scene/Auth/Login/Form.js

```diff
@@ -26,101 +33,141 @@ const _renderHeaderImage = () => (
   </View>
 );
 
-const _renderForm = () => (
-  <View style={styles.formContainer}>
-    <Field
-      name="email"
-      component={FormInput}
-      validate={[required, email]}
-      placeholder="Email"
-      underlineColorAndroid={'transparent'}
-      keyboardType={'email-address'}
-    />
-    <Field
-      name="password"
-      component={FormInput}
-      validate={[required, password]}
-      placeholder="Password"
-      underlineColorAndroid={'transparent'}
-      secureTextEntry={true}
-      returnKeyType={'done'}
-    />
-  </View>
-);
-
-const _renderButton = args => {
-  const { handleSubmit, onLogin, loading } = args;
-  return (
-    <View style={styles.buttonContainer}>
-      {loading ? (
-        _renderLoadingButton()
-      ) : (
-        <TouchableView
-          onPress={handleSubmit(onLogin)}
-          style={styles.submitButton}
-        >
-          <Text bold style={styles.buttonText}>
-            LOGIN
-          </Text>
-        </TouchableView>
-      )}
-    </View>
-  );
-};
-
-const _renderFooter = onNavigate => (
-  <View>
-    <TouchableView onPress={onNavigate}>
-      <Text style={styles.footerText}>Forgot your password.</Text>
-      <Text medium style={styles.signUpText}>
-        {"Don't have an account. Register"}
-      </Text>
-    </TouchableView>
-  </View>
-);
-
-const _renderError = error => (
-  <View>
-    {error === undefined || <Text style={styles.errorText}>{error}</Text>}
-  </View>
-);
-
 const LoginForm = ({
-  onLogin,
   onNavigate,
+  values,
   handleSubmit,
-  loading,
-  loginError,
+  setFieldValue,
+  setFieldTouched,
+  dirty,
+  isValid,
+  isSubmitting,
+  errors,
+  touched,
 }) => {
+  const isInvalid = dirty && !isValid;
   return (
     <KeyboardAvoidingView behavior={'padding'} style={styles.container}>
       {_renderHeaderImage()}
-      {_renderForm()}
-      {_renderError(loginError)}
-      {_renderButton({ handleSubmit, onLogin, loading })}
-      {_renderFooter(onNavigate)}
+      <View style={styles.formContainer}>
+        <FormLabel>Email</FormLabel>
+        <FormInput
+          name="email"
+          value={values.email}
+          onChangeText={text => {
+            setFieldValue('email', text);
```

Comment:
I can see you are repeating these line many times. How can you figure out when a form that is NOT simple to this Login form (e.g. User Profile have more than 10 input). We need to make the code look cleaner.